File: demo-app/app/src/javaApp/java/com/launchkey/android/authenticator/demo/ui/fragment/CustomLogoutFragment.java
package com.launchkey.android.authenticator.demo.ui.fragment;

import android.app.ProgressDialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.launchkey.android.authenticator.demo.R;
import com.launchkey.android.authenticator.demo.util.Utils;
import com.launchkey.android.authenticator.sdk.core.authentication_management.AuthenticatorManager;
import com.launchkey.android.authenticator.sdk.core.authentication_management.event_callback.EndAllSessionsEventCallback;

public class CustomLogoutFragment extends BaseDemoFragment {
    @NonNull
    private final AuthenticatorManager authenticatorManager = AuthenticatorManager.instance;

    private ProgressDialog logoutDialog;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        logoutDialog = new ProgressDialog(getActivity(), R.style.Theme_WhiteLabel_Dialog);
        logoutDialog.setIndeterminate(true);
        logoutDialog.setCancelable(false);
        logoutDialog.setMessage("Custom Logout...");

        logOut();
    }

    private void logOut() {

        Utils.show(logoutDialog);

        authenticatorManager.endAllSessions(new EndAllSessionsEventCallback() {
            @Override
            public void onSuccess(final Void result) {
                Utils.dismiss(logoutDialog);
                if (getActivity() != null && !getActivity().isFinishing()) {
                    getActivity().finish();
                }
            }

            @Override
            public void onFailure(final @NonNull Exception e) {
                Utils.dismiss(logoutDialog);
                String message = Utils.getMessageForBaseError(e);
                Utils.alert(getActivity(), "Error", message);
            }
        });
    }
}

File: demo-app/app/src/kotlinApp/java/com/launchkey/android/authenticator/demo/ui/fragment/CustomLogoutFragment.kt
package com.launchkey.android.authenticator.demo.ui.fragment

import android.app.ProgressDialog
import android.os.Bundle
import androidx.viewbinding.ViewBinding
import com.launchkey.android.authenticator.demo.R
import com.launchkey.android.authenticator.demo.util.Utils.alert
import com.launchkey.android.authenticator.demo.util.Utils.dismiss
import com.launchkey.android.authenticator.demo.util.Utils.getMessageForBaseError
import com.launchkey.android.authenticator.demo.util.Utils.show
import com.launchkey.android.authenticator.sdk.core.authentication_management.AuthenticatorManager
import com.launchkey.android.authenticator.sdk.core.authentication_management.event_callback.EndAllSessionsEventCallback

class CustomLogoutFragment : BaseDemoFragment<ViewBinding>() {
    private var logoutDialog: ProgressDialog? = null
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        logoutDialog = ProgressDialog(activity, R.style.Theme_WhiteLabel_Dialog)
        logoutDialog!!.isIndeterminate = true
        logoutDialog!!.setCancelable(false)
        logoutDialog!!.setMessage("Custom Logout...")
        logOut()
    }

    private fun logOut() {
        show(logoutDialog)
        authenticatorManager.endAllSessions(object : EndAllSessionsEventCallback() {
            override fun onSuccess(result: Void?) {
                dismiss(logoutDialog)
                if (activity != null && !activity!!.isFinishing) {
                    activity!!.finish()
                }
            }

            override fun onFailure(e: Exception) {
                dismiss(logoutDialog)
                val message = getMessageForBaseError(e)
                alert(activity, "Error", message)
            }
        })
    }
}